src/dirwatcher/index.ts

```diff
@@ -0,0 +1,70 @@
+import EventEmitter from 'events';
+import path from 'path';
+import fs from 'fs';
+import logger from '../../logger';
+import { DIRECTORY_FILES_CHANGED } from 'constants/EventNames';
+
+export default class DirWatcher extends EventEmitter {
+    private changedFiles = [];
+    private allFilesInDirectoryStats = {};
+
+    public watch(dirPath: string, delay: number) {
+        setInterval(() => this.readDirectory(dirPath), delay);
```

Comment:
it might make sense to return interval ID from watch function to clean up resources when needed.

Reply:
Ok